nisystemlink/clients/product/_product_client.py

```diff
@@ -0,0 +1,182 @@
+"""Implementation of Product Client"""
+
+from typing import List, Optional
+
+from nisystemlink.clients import core
+from nisystemlink.clients.core._uplink._base_client import BaseClient
+from nisystemlink.clients.core._uplink._methods import delete, get, post
+from nisystemlink.clients.product.models import Product
+from uplink import Field, Query, returns
+
+from . import models
+
+
+class ProductClient(BaseClient):
+    def __init__(self, configuration: Optional[core.HttpConfiguration] = None):
+        """Initialize an instance.
+
+        Args:
+            configuration: Defines the web server to connect to and information about
+                how to connect. If not provided, the
+                :class:`HttpConfigurationManager <nisystemlink.clients.core.HttpConfigurationManager>`
+                is used to obtain the configuration.
+
+        Raises:
+            ApiException: if unable to communicate with the Product Service.
+        """
+        if configuration is None:
+            configuration = core.HttpConfigurationManager.get_configuration()
+        super().__init__(configuration, base_path="/nitestmonitor/v2/")
+
+    @post("products", args=[Field("products")])
+    def create_products(
+        self, products: List[Product]
+    ) -> models.CreateProductsPartialSuccess:
+        """Creates one or more products and returns errors for failed creations.
+
+        Args:
+            products: A list of products to attempt to create.
+
+        Returns: A list of created products, products that failed to create, and errors for
+            failures.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` service of provided invalid
+                arguments.
+        """
+        ...
+
+    @get(
+        "products",
+        args=[Query("continuationToken"), Query("take"), Query("returnCount")],
+    )
+    def get_products(
```

Comment:
@adamarnesen , making this method `get_products_paged` will be appropriate as this could also become a generator.